File: Stack/JAVA/GenStack.java
import java.util.*;

class node<T>
{
    public T data;
    public node<T> next;
}

class StackX<T>
{
    private node<T> first;
    private int iCount;

    public StackX()
    {
        this.first = null;
        this.iCount = 0;
    }

    public void push(T no)
    {
        node<T> newn = null;

        newn = new node<T>();
        newn.data = no;
        newn.next = null;

        newn.next = first;
        first = newn;

        iCount++;
    }

    public T pop()
    {
        node<T> temp = null;

        if(first == null)
        {
            System.out.println("Unable to delete stack empty");
            return null;
        }
        else
        {
            temp = first;

            first = first.next;
            iCount--;

            return temp.data;
        }

    }

    public void Display()
    {
        node<T> temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.println("| "+temp.data+" |");
            temp = temp.next;
        }
    }

    public int Count()
    {
        return iCount;
    }
}

class GenStack
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        StackX<Integer> obj = new StackX<Integer>();
        int value = 0, choice = 0;

        System.out.println("===================================================");
        System.out.println("                   Stack Manager                   ");
        System.out.println("===================================================");

        while(true)
        {
            System.out.println("\n---------------------------------------------");
            System.out.println("Stack Operations Menu");
            System.out.println("---------------------------------------------");
            System.out.println("1 : Push element");
            System.out.println("2 : Pop element");
            System.out.println("3 : Display stack");
            System.out.println("4 : Count elements");
            System.out.println("5 : Exit");
            System.out.println("---------------------------------------------");
            System.out.print("Enter your choice: ");
            choice = sobj.nextInt();

            if(choice == 1)
            {
                System.out.print("Enter element to push: ");
                value = sobj.nextInt();
                obj.push(value);
                System.out.println("Element pushed successfully.");
            }
            else if(choice == 2)
            {
                value = obj.pop();
                if(value != -1)
                {
                    System.out.println("Popped element is: " + value);
                }
            }
            else if(choice == 3)
            {
                System.out.println("Current stack:");
                obj.Display();
            }
            else if(choice == 4)
            {
                value = obj.Count();
                System.out.println("Number of elements in stack: " + value);
            }
            else if(choice == 5)
            {
                System.out.println("Thank you for using stack manager.");
                break;
            }
            else
            {
                System.out.println("Invalid choice. Please try again.");
            }
        }

        sobj.close();
    }
}

File: Stack/CPP/GenStack.cpp
#include<iostream>
using namespace std;

template <class T>
struct node 
{
    T data;
    struct node<T> * next;
};

template <class T>
class Stack
{
    private:
        struct node<T> * first;
        int iCount;

    public:

        Stack()
        {
            this->first = NULL;
            this->iCount = 0;
        }

        void push(T no)
        {
            struct node<T> * newn = NULL;
            
            newn = new node<T>;
            newn->data = no;
            newn->next = NULL;

            newn->next = first;
            first = newn;

            iCount++;
        }

        T pop()
        {
            struct node<T> * temp = NULL;
            T value;

            if(first == NULL)
            {
                cout<<"Unable to delete stack empty\n";
                return NULL;
            }
            else
            {
                temp = first;

                first = first->next;
                value = temp->data;

                delete temp;
                iCount--;

                return value;
            }
        }

        void Display()
        {
            struct node<T> * temp;

            if(first == NULL)
            {
                cout << "Stack is empty\n";
            }

            temp = first;

            while (temp != NULL)
            {
                cout<<"| "<<temp->data<<" |\n";
                temp = temp->next;
            }
        }

        int Count()
        {
            return iCount;
        }
};

int main()
{
    Stack<int> * obj = new Stack<int>();

    int iRet = 0;
    int choice = 0;
    int value = 0;

    cout << "\n===================================" << endl;
    cout << "              Stack Manager          " << endl;
    cout << "=====================================\n";

    while (true)
    {
        cout << "\n-------------------------------" << endl;
        cout << "1 : Push element into stack" << endl;
        cout << "2 : Pop element from stack" << endl;
        cout << "3 : Display stack elements" << endl;
        cout << "4 : Count elements in stack" << endl;
        cout << "0 : Exit" << endl;
        cout << "-------------------------------" << endl;
        cout << "Enter your choice: ";
        cin >> choice;

        if (choice == 1)
        {
            cout << "Enter value to push: ";
            cin >> value;
            obj->push(value);
        }
        else if (choice == 2)
        {
            iRet = obj->pop();
            if (iRet != -1)
            {
                cout << "Popped element: " << iRet << "\n";
            }
        }
        else if (choice == 3)
        {
            obj->Display();
        }
        else if (choice == 4)
        {
            iRet = obj->Count();
            cout << "Number of elements: " << iRet << "\n";
        }
        else if (choice == 0)
        {
            cout << "Exiting application...\n";
            break;
        }
        else
        {
            cout << "Invalid choice. Please try again.\n";
        }
    }

    return 0;
}